// @mui/x-data-grid-generator/renderer/renderEditCurrency.js
"use strict";
'use client';

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.renderEditCurrency = renderEditCurrency;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var React = _interopRequireWildcard(require("react"));
var _Autocomplete = _interopRequireWildcard(require("@mui/material/Autocomplete"));
var _InputBase = _interopRequireDefault(require("@mui/material/InputBase"));
var _Box = _interopRequireDefault(require("@mui/material/Box"));
var _styles = require("@mui/material/styles");
var _staticData = require("../services/static-data");
var _useEditDropdownState = require("../hooks/useEditDropdownState");
var _jsxRuntime = require("react/jsx-runtime");
const StyledAutocomplete = (0, _styles.styled)(_Autocomplete.default)(({
  theme
}) => ({
  height: '100%',
  [`& .${_Autocomplete.autocompleteClasses.inputRoot}`]: (0, _extends2.default)({}, theme.typography.body2, {
    padding: '1px 0',
    height: '100%',
    '& input': {
      padding: '0 16px',
      height: '100%'
    }
  })
}));
function EditCurrency(props) {
  const {
    id,
    value,
    field,
    hasFocus
  } = props;
  const {
    open,
    setOpen,
    inputRef,
    shouldAutoOpen,
    handleAutocompleteInputKeyDown,
    handleAutocompleteWrapperKeyDown,
    createAutocompleteChangeHandler
  } = (0, _useEditDropdownState.useEditDropdownState)({
    id,
    field,
    hasFocus
  });
  const handleChange = createAutocompleteChangeHandler(val => val.toUpperCase());
  return (
    /*#__PURE__*/
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    (0, _jsxRuntime.jsx)("div", {
      onKeyDown: handleAutocompleteWrapperKeyDown,
      style: {
        height: '100%',
        width: '100%'
      },
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(StyledAutocomplete, {
        value: value,
        onChange: handleChange,
        options: _staticData.CURRENCY_OPTIONS,
        autoHighlight: true,
        fullWidth: true,
        open: open,
        onOpen: () => setOpen(true),
        onClose: () => setOpen(false),
        disableClearable: true,
        renderOption: (optionProps, option) => /*#__PURE__*/(0, _jsxRuntime.jsxs)(_Box.default, (0, _extends2.default)({
          component: "li",
          sx: {
            '& > img': {
              mr: 1.5,
              flexShrink: 0
            }
          }
        }, optionProps, {
          children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("img", {
            loading: "lazy",
            width: "20",
            src: `https://flagcdn.com/w20/${option.slice(0, -1).toLowerCase()}.png`,
            srcSet: `https://flagcdn.com/w40/${option.slice(0, -1).toLowerCase()}.png 2x`,
            alt: ""
          }), option]
        })),
        renderInput: params => /*#__PURE__*/(0, _jsxRuntime.jsx)(_InputBase.default, (0, _extends2.default)({
          autoFocus: shouldAutoOpen,
          fullWidth: true,
          id: params.id,
          inputRef: inputRef,
          onKeyDown: handleAutocompleteInputKeyDown,
          inputProps: (0, _extends2.default)({}, params.slotProps.htmlInput, {
            autoComplete: 'new-password' // disable autocomplete and autofill
          })
        }, params.slotProps.input))
      })
    })
  );
}
function renderEditCurrency(params) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(EditCurrency, (0, _extends2.default)({}, params));
}
