// @mui/x-data-grid-generator/renderer/renderEditBoolean.js
"use strict";
'use client';

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.renderEditBoolean = renderEditBoolean;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var React = _interopRequireWildcard(require("react"));
var _useEnhancedEffect = _interopRequireDefault(require("@mui/utils/useEnhancedEffect"));
var _xDataGridPremium = require("@mui/x-data-grid-premium");
var _useRowEditHandlers = require("../hooks/useRowEditHandlers");
var _jsxRuntime = require("react/jsx-runtime");
function EditBoolean(props) {
  const {
    id,
    value,
    field,
    hasFocus
  } = props;
  const apiRef = (0, _xDataGridPremium.useGridApiContext)();
  const rootProps = (0, _xDataGridPremium.useGridRootProps)();
  const inputRef = React.useRef(null);
  const {
    handleTabKeyDown
  } = (0, _useRowEditHandlers.useRowEditHandlers)({
    id,
    field
  });
  const handleChange = React.useCallback(async event => {
    await apiRef.current.setEditCellValue({
      id,
      field,
      value: event.target.checked
    }, event);
  }, [apiRef, field, id]);
  (0, _useEnhancedEffect.default)(() => {
    if (hasFocus) {
      inputRef.current?.focus();
    }
  }, [hasFocus]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
    onKeyDownCapture: handleTabKeyDown,
    style: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: '100%'
    },
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(rootProps.slots.baseCheckbox, (0, _extends2.default)({
      inputRef: inputRef,
      checked: Boolean(value),
      onChange: handleChange,
      size: "small"
    }, rootProps.slotProps?.baseCheckbox))
  });
}
function renderEditBoolean(params) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(EditBoolean, (0, _extends2.default)({}, params));
}
