// @mui/x-data-grid-generator/renderer/renderEditStatus.js
"use strict";
'use client';

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard").default;
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault").default;
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.renderEditStatus = renderEditStatus;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var React = _interopRequireWildcard(require("react"));
var _Select = _interopRequireDefault(require("@mui/material/Select"));
var _MenuItem = _interopRequireDefault(require("@mui/material/MenuItem"));
var _ListItemIcon = _interopRequireDefault(require("@mui/material/ListItemIcon"));
var _ListItemText = _interopRequireDefault(require("@mui/material/ListItemText"));
var _ReportProblem = _interopRequireDefault(require("@mui/icons-material/ReportProblem"));
var _Info = _interopRequireDefault(require("@mui/icons-material/Info"));
var _Autorenew = _interopRequireDefault(require("@mui/icons-material/Autorenew"));
var _Done = _interopRequireDefault(require("@mui/icons-material/Done"));
var _staticData = require("../services/static-data");
var _useEditDropdownState = require("../hooks/useEditDropdownState");
var _jsxRuntime = require("react/jsx-runtime");
function EditStatus(props) {
  const {
    id,
    value,
    field,
    hasFocus
  } = props;
  const {
    open,
    setOpen,
    inputRef,
    shouldAutoOpen,
    handleSelectKeyDown,
    handleSelectMenuClose,
    handleSelectMenuListKeyDown,
    createSelectChangeHandler
  } = (0, _useEditDropdownState.useEditDropdownState)({
    id,
    field,
    hasFocus
  });
  const handleChange = createSelectChangeHandler(event => event.target.value);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_Select.default, {
    value: value,
    onChange: handleChange,
    onKeyDown: handleSelectKeyDown,
    onOpen: () => setOpen(true),
    inputRef: inputRef,
    MenuProps: {
      onClose: handleSelectMenuClose,
      slotProps: {
        list: {
          onKeyDown: handleSelectMenuListKeyDown
        }
      }
    },
    sx: {
      height: '100%',
      '& .MuiSelect-select': {
        display: 'flex',
        alignItems: 'center',
        pl: 1
      }
    },
    autoFocus: shouldAutoOpen,
    fullWidth: true,
    open: open,
    children: _staticData.STATUS_OPTIONS.map(option => {
      let IconComponent = null;
      if (option === 'Rejected') {
        IconComponent = _ReportProblem.default;
      } else if (option === 'Open') {
        IconComponent = _Info.default;
      } else if (option === 'Partially Filled') {
        IconComponent = _Autorenew.default;
      } else if (option === 'Filled') {
        IconComponent = _Done.default;
      }
      return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_MenuItem.default, {
        value: option,
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_ListItemIcon.default, {
          sx: {
            minWidth: 36
          },
          children: /*#__PURE__*/(0, _jsxRuntime.jsx)(IconComponent, {
            fontSize: "small"
          })
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_ListItemText.default, {
          primary: option,
          sx: {
            overflow: 'hidden'
          }
        })]
      }, option);
    })
  });
}
function renderEditStatus(params) {
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(EditStatus, (0, _extends2.default)({}, params));
}
