// @mui/x-license/internals/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  base64Decode: true,
  base64Encode: true,
  md5: true,
  APP_TYPES: true,
  PLAN_SCOPES: true,
  PLAN_VERSIONS: true,
  LICENSE_MODELS: true,
  Watermark: true,
  useLicenseVerifier: true,
  clearLicenseStatusCache: true
};
Object.defineProperty(exports, "APP_TYPES", {
  enumerable: true,
  get: function () {
    return _licenseAppType.APP_TYPES;
  }
});
Object.defineProperty(exports, "LICENSE_MODELS", {
  enumerable: true,
  get: function () {
    return _licenseModel.LICENSE_MODELS;
  }
});
Object.defineProperty(exports, "PLAN_SCOPES", {
  enumerable: true,
  get: function () {
    return _licensePlan.PLAN_SCOPES;
  }
});
Object.defineProperty(exports, "PLAN_VERSIONS", {
  enumerable: true,
  get: function () {
    return _licensePlan.PLAN_VERSIONS;
  }
});
Object.defineProperty(exports, "Watermark", {
  enumerable: true,
  get: function () {
    return _Watermark.Watermark;
  }
});
Object.defineProperty(exports, "base64Decode", {
  enumerable: true,
  get: function () {
    return _base.base64Decode;
  }
});
Object.defineProperty(exports, "base64Encode", {
  enumerable: true,
  get: function () {
    return _base.base64Encode;
  }
});
Object.defineProperty(exports, "clearLicenseStatusCache", {
  enumerable: true,
  get: function () {
    return _useLicenseVerifier.clearLicenseStatusCache;
  }
});
Object.defineProperty(exports, "md5", {
  enumerable: true,
  get: function () {
    return _md.md5;
  }
});
Object.defineProperty(exports, "useLicenseVerifier", {
  enumerable: true,
  get: function () {
    return _useLicenseVerifier.useLicenseVerifier;
  }
});
var _base = require("../encoding/base64");
var _md = require("../encoding/md5");
var _licenseAppType = require("../utils/licenseAppType");
var _licensePlan = require("../utils/licensePlan");
var _licenseModel = require("../utils/licenseModel");
var _Watermark = require("../Watermark/Watermark");
var _useLicenseVerifier = require("../useLicenseVerifier/useLicenseVerifier");
var _verifyLicense = require("../verifyLicense/verifyLicense");
Object.keys(_verifyLicense).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _verifyLicense[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _verifyLicense[key];
    }
  });
});
var _testKeys = require("../test-keys");
Object.keys(_testKeys).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _testKeys[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _testKeys[key];
    }
  });
});
